hypex/dataset/abstract.py

```diff
@@ -100,16 +100,31 @@ def __len__(self):
         return self._backend.__len__()
 
     def get_columns_by_roles(
```

Comment:
Do you want to rename `get_columns_by_roles` to `search_columns`? And delete `get_columns_by_type`, because this function is universal?